src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIResponseChatClient.cs

```diff
@@ -0,0 +1,582 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Shared.Diagnostics;
+using OpenAI.Responses;
+
+#pragma warning disable S1067 // Expressions should not be too complex
+#pragma warning disable S3011 // Reflection should not be used to increase accessibility of classes, methods, or fields
+#pragma warning disable SA1204 // Static elements should appear before instance elements
+#pragma warning disable SA1108 // Block statements should not contain embedded comments
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>Represents an <see cref="IChatClient"/> for an <see cref="OpenAIResponseClient"/>.</summary>
+internal sealed class OpenAIResponseChatClient : IChatClient
```

Comment:
Having this be `internal` makes sense. However `OpenAIChatClient` is `public` - should we update that to be `internal` or do you think there's a reason the two should differ? (Note: `OpenAIAssistantClient` is `internal`).

Reply:
Yup. I didn't want to do it as part of this PR, though. It's tracked by https://github.com/dotnet/extensions/issues/6094.